states/new_hampshire/main.py

```diff
@@ -0,0 +1,84 @@
+from common import work_dir
+import json
+import scrapy
+from scrapy.crawler import CrawlerProcess
+import csv
+import re
+from io import StringIO
+import numpy as np
+
+# We use Scrapy for New Hampshire because it has an .aspx __viewstate which must be generated and sent along with the request
+
+
+class NewHampshireSpider(scrapy.Spider):
+    name = 'New Hampshire'
+    start_urls = ["https://app.sos.nh.gov/Public/Reports.aspx"]
+
+    def parse(self, response):
+        yield scrapy.FormRequest.from_response(
+            response,
+            formdata={
+                'ctl00$MainContentPlaceHolder$PPReport': 'rdoCsv'
+            },
+            method="POST",
+            callback=self.generate_csv
+        )
+
+    def generate_csv(self, response):
+        f = StringIO(response.text)
+        reader = csv.DictReader(f, delimiter=",")
+        clerk_data = []
+        for row in reader:
+            town_or_city = " ".join([x.lower().capitalize()
+                                     for x in row["Town/City"].split(" ")])
+            email_pattern = re.compile(
+                r"^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$")
+
+            if(not email_pattern.match(row["E-Mail"]) or not row["Fax"]):
```

Comment:
Just because a row doesn't have a fax or email, doesn't mean we the other data doesn't exist.  We should eagerly load what we can and use `[]` when there's no field available.